src/NzbDrone.Core/Qualities/QualityDefinitionLimits.cs

```diff
@@ -0,0 +1,12 @@
+﻿using System.Diagnostics.CodeAnalysis;
+
+namespace NzbDrone.Core.Qualities;
+
+[SuppressMessage("Performance", "CA1822:Mark members as static", Justification =
```

Comment:
We should be able to serialize the static values, probably just need to tell the serializer how, though just having these as static values and translating to a resource in the API would be the typical way we've been doing this.

Reply:
I do not believe they should be static. Perhaps I could enhance the wording of my justification here, but my goal was to silence the false positive here. I don't personally like leaving analysis issues for the next person...

Reply:
Why shouldn't they be static? They're not going to be changed during run time and assuming validation happens on the API side they'd be referenced there as well instead of just serialized to JSON.

Perhaps though they'd be better as constants and the API can generate the expected object.

Reply:
I'm thinking of it the other way around:

- Why *should* they be static?
- Does the difference matter?
- What's wrong with how they currently are?

I'd rather spend cycles working on your other feedback which is far more important, IMO.

Reply:
I'm going to decline the request to change these to static. Out of respect for your request, I did do a little more research just to make sure my opinion is informed. Here are the reasons I'm going to keep it as-is:

- These properties define the surface of a public API. The fact that the values are constant is an implementation detail. They may not always be constant. From a design standpoint, these properties make more sense as non-static given the intent and their meaning, as well as the context in which they are used (DTO, API contract).
- There doesn't seem to be an easy way to serialize static members. I'm not saying it's impossible, but it's just not easy. It's just quite frankly not the right solution for serializable types (i.e. DTOs). The pattern I've adopted here (instance properties) is very typical for DTOs.

Having said that, I have made the justification message less confusing, so hopefully that helps. That will be available in my next push.

Reply:
> Because they never change during run time, regardless of them changing in the future they will never change during run time, which a static value is perfect for. It would also allow another consumption of these values without creating another instance of these limits.

I appreciate and understand your reasoning, however for the reasons previously mentioned I'm going to keep it as-is. Consumption of the limits elsewhere is currently not a requirement and so is not something I'm willing to consider here; it would be a simple matter to refactor this code later, should that become a requirement.

> Defining a class within Core and then using that directly as a resource in the API is not something that we typically do because the API contract doesn't necessarily follow the internal structure; albeit unlikely in this particular instance.

This reframes the discussion a bit. My goal was to reduce maintenance overhead by avoiding duplicating this class. If you insist on this anyway, then your argument for static makes more sense -- in that case, the `Resource` version would be non-static (for the reasons I mentioned in my last reply) and the non-resource version in Core would probably be static since I can see how no instance is really needed there, although I'll admit because I skipped this pattern I did not look close enough to ensure that's an informed opinion.

> You were talking about validation previously, in validation we wouldn't create a new instance of this DTO in order to validate that each quality is within the correct bounds.

I'm not quite sure what your specific expectations are regarding tests for validation; however in what I've started, the DTO is indeed created multiple times for validation purposes, but in the context of a test I'm not sure how or why this matters. Semantically, a new instance is only created once per test method, but that's required because I do not typically share state at class scope in test fixtures as a personal policy. Again, I'm willing to follow the direction of the project here; what I've started really was just in lieu of understanding a desired approach.

I think if you can let me know how strongly you feel about duplicating the class (to have a separate resource version for the API and another for the core domain), that will ultimately determine the outcome of the static discussion (which isn't the main focus as I'm coming to understand).

Reply:
I don't feel too strongly about separating the core from API in this instance, but once it's used in multiple places I think it being a static variable will make more sense. If you have a solution in for validation I'm fine to defer this until you have an example for that.

Reply:
As commented on the usage of the validator I think it makes sense to move to static values here and a resource class to handle the serialization on the API side.

This will also allow the limits to be used in quality profiles without refactoring.